apps/medon-fe/src/components/AvailabilityCalendar/utils.ts

```diff
@@ -0,0 +1,111 @@
+import dayjs from 'dayjs';
+import { Event } from 'react-big-calendar';
+
+import { IAvailability } from 'redux/api/types';
+import { timeFormat } from 'utils/constants';
+import { endOfDayHour } from 'utils/constants/options/hourOptions';
+import { CalendarSlot } from './types';
+
+export const convertSlotToArray = (timeSlot: CalendarSlot) => {
+  const startHour = dayjs(timeSlot.start).hour();
+  const endHour =
+    dayjs(timeSlot.end).hour() !== 0
+      ? dayjs(timeSlot.end).hour()
+      : endOfDayHour;
+  const availabilityArray = [];
+
+  if (endHour - startHour === 1) {
+    return [
+      {
+        startTime: dayjs(timeSlot.start).toDate(),
+        endTime: dayjs(timeSlot.end).toDate(),
+        title: timeSlot.title,
+      },
+    ];
+  }
+  for (let i = startHour; i < endHour; i += 1) {
+    const startTime = dayjs(timeSlot.start).hour(i).toDate();
+
+    availabilityArray.push({
+      startTime,
+      endTime: dayjs(startTime).add(1, 'hour').toDate(),
+      title: timeSlot.title,
+    });
+  }
+
+  return availabilityArray;
+};
+
+export const joinConsecutiveDates = (dates: IAvailability[]) => {
```

Comment:
where is output type?

Reply:
left some comments before the function which explain what does it do, its look quet difficult

Reply:
return types added, comment added too